pkg/generate/code/set_resource.go

```diff
@@ -110,23 +110,14 @@ func SetResource(
 	if op == nil {
 		return ""
 	}
-	outputShape := op.OutputRef.Shape
+	outputShape, _ := r.GetOutputShape(op)
 	if outputShape == nil {
 		return ""
 	}
 
-	var err error
-	// We might be in a "wrapper" shape. Unwrap it to find the real object
-	// representation for the CRD's createOp/DescribeOP.
-
 	// Use the wrapper field path if it's given in the ack-generate config file.
 	wrapperFieldPath := r.GetOutputWrapperFieldPath(op)
 	if wrapperFieldPath != nil {
-		outputShape, err = r.GetWrapperOutputShape(outputShape, *wrapperFieldPath)
-		if err != nil {
-			msg := fmt.Sprintf("Unable to unwrap the output shape: %v", err)
-			panic(msg)
-		}
 		sourceVarName += "." + *wrapperFieldPath
 	} else {
 		// If the wrapper field path is not specified in the config file and if
```

Comment:
Sorry, but why didn't this block of code pick up the `VpcEndpoint` shape as being the wrapper struct?

Reply:
This code did work as intended; however, because `model.go:164` executes first and **was not** checking for overrides initially, the code-generator would crash because it could not locate `VpcEndpoint` identifier. 

I added `GetOutputShape` call here after refactoring to clean up the code a bit-- the logic remains the same.

Reply:
I would like to clarify that the original use for `output_wrapper_field_path` was working: You could unwrap a describe output structure into status and spec fields. 

What we are adding in this is support for `output_wrapper_field_path` on a create output. That is, taking a wrapper structure from a create and creating new status and spec fields from its members. [CreateVpcEndpointOutput](https://docs.aws.amazon.com/sdk-for-go/api/service/ec2/#CreateVpcEndpointOutput) has a wrapper of `VpcEndpoint` - its members are what we want in the spec and status for the CRD.

Some of the code changes @brycahta has made are re-using existing parts of the code that worked with the describe output to also work with create. I believe this is one of those elements.